### PR TITLE
FSET-1738 - Added clearCandidates to test data generator

### PR DESCRIPTION
**Pull request by andrewrembrandt**

### CODE REVIEW

**Comment by fayimora** [comment]
This clears only the application but not the corresponding user from auth provider.

File: app/repositories/application/GeneralApplicationRepository.scala [hunk 1]
```diff
@@ -907,6 +910,12 @@ class GeneralApplicationMongoRepository(
     replaceAllocationStatus(applicationId, EventProgressStatuses.get(eventType.applicationStatus).failedToAttend)
   }
 
+  override def remove(applicationStatus: Option[String]): Future[Int] = {
+    val query = applicationStatus.map(as => BSONDocument("applicationStatus" -> as)).getOrElse(BSONDocument())
```
Comment (fayimora) [inline app/repositories/application/GeneralApplicationRepository.scala]: I'll prefer to call this `appStatus` or `status`.

**Comment by andrewrembrandt** [comment]
I've added the call to Auth Provider to delete - just need to test quickly once I'm in the office

**Comment by henricook** [comment]
What's your use case for this?

**Comment by henricook** [comment]
Can you plaster a warning comment on both the authproviderclient method and generalapplicationrepository method saying this is for test data reasons only, we don't want anyone ever using these methods by accident in features

**Comment by henricook** [comment]
I notice the DELETE endpoint in the auth provider isn't in test only routes which would make me feel beter - it was added by @mgilgar  - Miguel do we need that endpoint to be a production endpoint for some feature?

**Comment by andrewrembrandt** [comment]
DELETE was added by @mgilgar when he did the CreateAssessor work in June - it's used in the admin F/E when user creation failed on the backend (i.e. it removes the associated user), so is required in production.

All tests pass, @kjharkin, ready for merge (with the admin AT PR also)

File: app/repositories/application/GeneralApplicationRepository.scala (hunk 1 above)
Comment (andrewrembrandt) [inline app/repositories/application/GeneralApplicationRepository.scala]: So, that in fact totally opposite of what we should be doing - the Robert C Martin, books, LSCC, and top consultants advocate use of full naming - i.e. no ambiguity

Comment (henricook) [inline app/repositories/application/GeneralApplicationRepository.scala]: I'm going to come out on the side of full names

**Comment by henricook** [comment]
Something about having a removeAllUsersInStatus method available to be used makes me really nervous

**Comment by henricook** [comment]
Is it worth adding a feature flag to ensure this can *never* be used accidentally in production @kjharkin , or am i overreacting?

**Comment by kjharkin** [comment]
Can we move this remove to a specific Test.... repository please.

**Comment by andrewrembrandt** [comment]
@kjharkin, can't put it into 'test' or 'it' scope as it's used via TestDataGenerator, so I've moved 'remove' into a dedicated repo that exists in a 'testdata' package which matches with the TDG package.

**Comment by henricook** [comment]
:muscle: - you had a conflict though

**Pull request merged**